### PR TITLE
switch-mupdf: add package

### PR DESCRIPTION
**Pull request by uyjulian**

### CODE REVIEW

**Comment by WinterMute** [comment]
Changed your mind about having this?

**Comment by uyjulian** [comment]
I'm going to make some changes to reduce the initial size footprint at a later time, and open a new PR.

**Comment by fincs** [comment]
You don't need to make a new PR, you can just push or forcepush to your `switch-mupdf` branch.

**Comment by uyjulian** [comment]
I decided to not make any changes, since it works fine.

File: switch/mupdf/mupdf.patch [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ 	FILE *file;
+ 	unsigned char buffer[4096];
++	const char *file_name;
+ } fz_file_stream;
+ 
+ static int next_file(fz_context *ctx, fz_stream *stm, size_t n)
+ {
+ 	fz_file_stream *state = stm->state;
+ 
++	/* FIXME: Multiple calls to fseek/fread return incorrect part of file */
```
Comment (WinterMute) [inline switch/mupdf/mupdf.patch:15]: This should be fixed now I hope. We corrected an issue with file buffering with devkitA64 r15

File: switch/mupdf/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  make generate
+
+  source /opt/devkitpro/switchvars.sh
+
+  make build=release \
+    CC=${TOOL_PREFIX}gcc CXX=${TOOL_PREFIX}g++ AR=${TOOL_PREFIX}ar \
+    LD=${TOOL_PREFIX}ld RANLIB=${TOOL_PREFIX}ranlib \
+    XCFLAGS="${CPPFLAGS} '-Dtimegm(x)=-1'" \
+    OS=switch \
```
Comment (WinterMute) [inline switch/mupdf/PKGBUILD:36]: Would it not be better to patch Makerules and do all this configuration for switch there?

http://git.ghostscript.com/?p=mupdf.git;a=blob;f=Makerules;h=d2e585daa0708ea23b961bf39dc407e9733744f1;hb=refs/heads/master#l168

**Comment by WinterMute** [comment]
Have you lost interest in fixing this & getting it added to packages?

**Comment by uyjulian** [comment]
No, I still plan to fix this myself.

**Comment by uyjulian** [comment]
I applied suggestions and updated to 1.17.0.  
I also made sure only the libraries are installed.

**Pull request merged**